distributed/tests/test_client.py

```diff
@@ -6062,15 +6041,24 @@ def f():
 
 
 @gen_cluster()
-async def test_mixing_clients(s, a, b):
+async def test_mixing_clients_same_scheduler(s, a, b):
     async with Client(s.address, asynchronous=True) as c1, Client(
         s.address, asynchronous=True
     ) as c2:
         future = c1.submit(inc, 1)
-        with pytest.raises(ValueError):
-            c2.submit(inc, future)
+        assert await c2.submit(inc, future) == 3
+    assert not s.tasks
```

Comment:
This is new. I simplified the serialization of the future and if the key is known to the scheduler, a future may even be shared by clients.
This is not necessary for this change just something I noticed when playing with the futures reducer.
Different scheduler still raises, see below (but only on await)